cmd/tests/tracing_module_test.go

```diff
@@ -128,6 +98,228 @@ func TestTracingClient_DoesNotInterfereWithHTTPModule(t *testing.T) {
+
+	tb.Mux.HandleFunc("/tracing", func(w http.ResponseWriter, r *http.Request) {
+		atomic.AddInt64(&gotRequests, 1)
+		assert.NotEmpty(t, r.Header.Get("traceparent"))
+		assert.Len(t, r.Header.Get("traceparent"), 55)
+	})
+
+	script := tb.Replacer.Replace(`
+		import http from "k6/http";
+		import tracing from "k6/experimental/tracing";
+
+		tracing.instrumentHTTP({
+			propagator: "w3c",
+		})
+
+		export default function () {
+			http.del("HTTPBIN_IP_URL/tracing");
+			http.get("HTTPBIN_IP_URL/tracing");
+			http.head("HTTPBIN_IP_URL/tracing");
+			http.options("HTTPBIN_IP_URL/tracing");
+			http.patch("HTTPBIN_IP_URL/tracing");
+			http.post("HTTPBIN_IP_URL/tracing");
+			http.put("HTTPBIN_IP_URL/tracing");
+			http.request("GET", "HTTPBIN_IP_URL/tracing");
+		};
+	`)
+
+	ts := getSingleFileTestState(t, script, []string{"--out", "json=results.json"}, 0)
+	cmd.ExecuteWithGlobalState(ts.GlobalState)
+
+	assert.Equal(t, int64(8), atomic.LoadInt64(&gotRequests))
+
+	jsonResults, err := afero.ReadFile(ts.FS, "results.json")
+	require.NoError(t, err)
+
+	assertHasTraceIDMetadata(t, jsonResults)
+}
+
+func TestTracingInstrumentHTTP_Jaeger(t *testing.T) {
+	t.Parallel()
+	tb := httpmultibin.NewHTTPMultiBin(t)
+
+	var gotRequests int64
+
+	tb.Mux.HandleFunc("/tracing", func(w http.ResponseWriter, r *http.Request) {
+		atomic.AddInt64(&gotRequests, 1)
+		assert.NotEmpty(t, r.Header.Get("uber-trace-id"))
+		assert.Len(t, r.Header.Get("uber-trace-id"), 45)
+	})
+
+	script := tb.Replacer.Replace(`
+		import http from "k6/http";
+		import { check } from "k6";
+		import tracing from "k6/experimental/tracing";
+
+		tracing.instrumentHTTP({
+			propagator: "jaeger",
+		})
+
+		export default function () {
+			http.del("HTTPBIN_IP_URL/tracing");
+			http.get("HTTPBIN_IP_URL/tracing");
+			http.head("HTTPBIN_IP_URL/tracing");
+			http.options("HTTPBIN_IP_URL/tracing");
+			http.patch("HTTPBIN_IP_URL/tracing");
+			http.post("HTTPBIN_IP_URL/tracing");
+			http.put("HTTPBIN_IP_URL/tracing");
+			http.request("GET", "HTTPBIN_IP_URL/tracing");
+		};
+	`)
+
+	ts := getSingleFileTestState(t, script, []string{"--out", "json=results.json"}, 0)
+	cmd.ExecuteWithGlobalState(ts.GlobalState)
+
+	assert.Equal(t, int64(8), atomic.LoadInt64(&gotRequests))
+
+	jsonResults, err := afero.ReadFile(ts.FS, "results.json")
+	require.NoError(t, err)
+
+	assertHasTraceIDMetadata(t, jsonResults)
+}
+
+func TestTracingInstrumentHTTP_FillsParams(t *testing.T) {
+	t.Parallel()
+	tb := httpmultibin.NewHTTPMultiBin(t)
+
+	var gotRequests int64
+
+	tb.Mux.HandleFunc("/tracing", func(w http.ResponseWriter, r *http.Request) {
+		atomic.AddInt64(&gotRequests, 1)
+
+		assert.NotEmpty(t, r.Header.Get("traceparent"))
+		assert.Len(t, r.Header.Get("traceparent"), 55)
+
+		assert.NotEmpty(t, r.Header.Get("X-Test-Header"))
+		assert.Equal(t, "test", r.Header.Get("X-Test-Header"))
+	})
+
+	script := tb.Replacer.Replace(`
+		import http from "k6/http";
+		import { check } from "k6";
```

Comment:
```suggestion
```
@oleiade there are multiple lines of it

Reply:
Good catch! :eyes: I guess this is the downside of writing JS in a Go string, no eslint, no prettier, no checks whatsoever.